pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -13,8 +13,10 @@
 # limitations under the License
 import logging
 import os
-from typing import Optional, Tuple
+from typing import Optional, Tuple, TYPE_CHECKING
 
+if TYPE_CHECKING:
+    from pytorch_lightning import Trainer
```

Comment:
this won't work. Use `import pytorch_lightning as pl` instead and annotate with `'pl.Trainer'`

Reply:
@justusschock can you please mention the reason why it won't work?

Reply:
sphinx cannot deal with the forward references you provide here as a result from the optional import